pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -69,7 +70,13 @@ def on_trainer_init(
         # it is important that these are the last callbacks to run
         self.trainer.callbacks = self._reorder_callbacks(self.trainer.callbacks)
 
-    def configure_checkpoint_callbacks(self, checkpoint_callback: Union[ModelCheckpoint, bool]):
+    def _configure_checkpoint_callbacks(self, checkpoint_callback: bool) -> None:
+        if not isinstance(checkpoint_callback, bool):
```

Comment:
Why do we want to explicitly check this? We don't do it for other properties. Correct typing should be enough for users to see that they are passing the wrong argument type

Reply:
@carmocca purely to alert users since this was previously supported behavior. I can add a TODO to remove this in v1.5 so we rely purely on the type signature. wdyt?